File: core/lib/widgets/Chart/js/chart.js
(function () {
    if (typeof BarChart === 'undefined')
        window.BarChart = {};

    BarChart = BarChart.BarChart = function (data) {
        this.data = data
    }

    BarChart.prototype.init = function (type) {
        let wrapper = document.createElement('div')
        wrapper.setAttribute('id', 'chart')
        let canvas = document.createElement('canvas')
        canvas.setAttribute('id', 'bar-chart')
        wrapper.append(canvas)
        document.getElementById('result').append(wrapper)
        const ctx = canvas.getContext('2d');
        let data = JSON.parse(this.data)
        new Chart(ctx, {
            type: type,
            data: data,
            options: {
                scales: {
                    y: {
                        beginAtZero: true
                    }
                }
            }
        });
    }
})();

File: core/lib/widgets/Chart/chart.py
from RenderElementInterface import RenderElementInterface
import js
import configparser
from js import BarChart
from pyodide.ffi import create_proxy
import json


class chart(RenderElementInterface):
    def __init__(self, data, type, category):
        self._data = data
        self._category = category
        self._type = type

    def render(self) -> None:
        bar_chart_class = create_proxy(BarChart.new(json.dumps(self._data['data'])))
        proxy_init = create_proxy(bar_chart_class.init(str(self._type)))
        proxy_init.destroy()
        bar_chart_class.destroy()
